scripts/update_and_rebuild_mfem.sh

```diff
@@ -0,0 +1,125 @@
+  echo "--skip-submodule-update  Do not update the libMesh submodule, use the current version"
+  echo
+  echo "Influential variables"
+  echo "CONDUIT_SRC_DIR          Path to conduit; default: ../framework/contrib/conduit from submodule"
+  echo "HDF5_DIR                 Path to hdf5; default PETSC_DIR"
+  echo "HYPRE_DIR                Path to hypre; default PETSC_DIR"
+  echo "LIBMESH_DIR              Path to libmesh (for netcdf); default ../libmesh/installed"
+  echo "METIS_DIR                Path to metis; default PETSC_DIR"
+  echo "MFEM_DIR                 MFEM install prefix; default: ../framework/contrib/mfem/installed"
+  echo "MFEM_SRC_DIR             Path to MFEM source; default: ../framework/contrib/mfem from submodule"
+  echo "PETSC_ARCH               PETSc arch; default: arch-moose if PETSC_DIR not set"
+  echo "PETSC_DIR                Path to PETSc install; default: ../petsc"
+  exit 0
+fi
+
+set -e
+
+if [ -n "$MFEM_SRC_DIR" ]; then
+  skip_sub_update=1
+else
+  MFEM_SRC_DIR="$(realpath -m "${SCRIPT_DIR}"/../framework/contrib/mfem)"
+fi
+MFEM_BUILD_DIR="${MFEM_SRC_DIR}/build"
+if [ -n "$MFEM_DIR" ]; then
+  echo "INFO: MFEM_DIR set - overriding default installed path"
+  echo "INFO: No cleaning will be done in specified path"
+else
+  MFEM_DIR="${MFEM_SRC_DIR}/installed"
+  rm -rf "$MFEM_DIR"
+fi
+
+CONDUIT_BUILD_DIR="${CONDUIT_SRC_DIR}/build"
+LIBMESH_DIR=${LIBMESH_DIR:-$(realpath -m "${SCRIPT_DIR}/../libmesh/installed")}
+if [ -z "$PETSC_DIR" ]; then
+  PETSC_DIR=$(realpath -m "${SCRIPT_DIR}/../petsc/arch-moose")
+fi
+HDF5_DIR=${HDF5_DIR:-$PETSC_DIR}
+HYPRE_DIR=${HYPRE_DIR:-$PETSC_DIR}
+METIS_DIR=${METIS_DIR:-$PETSC_DIR}
+
+if [ -z "$skip_sub_update" ]; then
+  cd "${SCRIPT_DIR}/.."
+  git submodule update --init --checkout framework/contrib/conduit
+  git submodule update --init --checkout framework/contrib/mfem
+  cd framework/contrib/conduit
+  git submodule update --init
+fi
+
+# Clean up old builds
+rm -rf "$CONDUIT_BUILD_DIR" "$MFEM_BUILD_DIR"
+
+# Build and install conduit
+mkdir -p "$CONDUIT_BUILD_DIR"
+cd "$CONDUIT_BUILD_DIR"
+cmake ../src -DCMAKE_INSTALL_PREFIX="$MFEM_DIR"
+make -j "${MOOSE_JOBS:-4}"
+make install
+
+# Build and install mfem
+mkdir -p "$MFEM_BUILD_DIR"
+cd "$MFEM_BUILD_DIR"
+cmake .. \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=YES \
+    -DMFEM_USE_OPENMP=NO \
+    -DMFEM_THREAD_SAFE=NO \
+    -DHYPRE_DIR="$HYPRE_DIR" \
+    -DMFEM_USE_MPI=YES \
+    -DMFEM_USE_METIS=YES \
+    -DMFEM_USE_METIS_5=YES \
+    -DMETIS_DIR="$METIS_DIR" \
+    -DParMETIS_DIR="$HYPRE_DIR" \
+    -DMFEM_USE_SUPERLU=YES \
+    -DSuperLUDist_DIR="$HYPRE_DIR" \
+    -DBUILD_SHARED_LIBS=ON \
+    -DHDF5_DIR="$HDF5_DIR" \
+    -DBLAS_DIR="$HYPRE_DIR" \
+    -DMFEM_ENABLE_EXAMPLES=yes \
+    -DMFEM_ENABLE_MINIAPPS=yes \
+    -DBLAS_LIBRARIES="$HYPRE_DIR/lib/libfblas.a" \
+    -DLAPACK_LIBRARIES="$HYPRE_DIR/lib/libflapack.a" \
+    -DBLAS_INCLUDE_DIRS="$HYPRE_DIR/include" \
+    -DLAPACK_INCLUDE_DIRS="$HYPRE_DIR/include" \
```

Comment:
If not making HYPRE/METIS/HDF5_DIR configurable, we can just set everything with the PETSc variables

Reply:
Removed